athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java

```diff
@@ -248,11 +250,60 @@ public GetSplitsResponse doGetSplits(BlockAllocator blockAllocator, GetSplitsReq
         return new GetSplitsResponse(getSplitsRequest.getCatalogName(), splits, null);
     }
 
+    @Override
+    protected List<TableName> listTables(final Connection jdbcConnection, final String databaseName)
+            throws SQLException
+    {
+        ImmutableList.Builder<TableName> list = ImmutableList.builder();
+
+        // Gets list of Tables and Views using Information Schema.tables
+        list.addAll(JDBCUtil.getTables(jdbcConnection, databaseName));
+        // Gets list of Materialized Views using table pg_catalog.pg_matviews 
+        list.addAll(getMaterializedViews(jdbcConnection, databaseName));
+
+        return list.build();
+    }
+
+    private String caseSensitiveNameResolver(PreparedStatement preparedStatement, String tableName, String databaseName) throws SQLException
+    {
+        String resolvedName = null;
+        try (ResultSet resultSet = preparedStatement.executeQuery()) {
+            if (resultSet.next()) {
+                resolvedName = resultSet.getString("table_name");
```

Comment:
MV SQL return this, `TABLE_NAME`. but here we are looking result set for table_name.
```
  String sql = "select matviewname as \"TABLE_NAME\" from pg_catalog.pg_matviews mv where (matviewname = ? or lower(matviewname) = ?) and schemaname = ?";
```

Reply:
[Documentation](https://docs.oracle.com/javase/tutorial/jdbc/basics/retrieving.html#:~:text=Strings%20used%20as%20input%20to%20getter%20methods%20are%20case%2Dinsensitive) states that "Strings used as input to getter methods are case-insensitive"